atlp-rwanda/space-bn-backend
# Ft signup #175743313

### What does this PR do?
- Implement user signup and generate a token.
### Description of Task to be completed?
- Implement the signup.
- Implement the JWT token generation
### How should this be manually tested?
- Clone the repository.
- In your home directory  move  into the cloned folder by **cd ft-signup-#175743313**
- Then run **npm install** to install all dependencies.
- Run **npm run migrate** to run the migrations of the user model
- Run **npm start** to start the server.
- access the swagger documentation on: **localhost:5000/api-docs**
-  after successful login u can use a middleware "check-auth.js" to protect routes with JWT
- Run **npm run test** to run the test.
### Any background context you want to provide?
- N/A
### What are the relevant Pivotal Tracker stories?
- [#175743313](https://www.pivotaltracker.com/story/show/175743313)
### Screenshots (if appropriate)
- N/A
### Questions
- N/A
